reactor-core/src/main/java/reactor/core/publisher/Mono.java

```diff
@@ -1624,7 +1625,7 @@ public T block(Duration timeout) {
 	 * @see #materialize()
 	 * @see Signal
 	 */
-	public final Mono<T> doOnEach(Consumer<? super Signal<T>> signalConsumer) {
+	public final Mono<@NonNull T> doOnEach(Consumer<? super @NonNull Signal<@NonNull T>> signalConsumer) {
```

Comment:
doubt on Signal's `<T>` non-nullability (consider the case of a `Signal.complete()`, its `T get()` method would return `null`)